pkg/webhook/validators.go

```diff
@@ -0,0 +1,56 @@
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	cv1alpha1 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	pv1alpha1 "github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	gvk := metav1.GroupVersionKind{
+		Kind:    "Egress",
+		Group:   "policy.osm.io",
+		Version: "v1alpha1",
+	}
+	RegisterValidator(gvk.String(), EgressValidator)
+	RegisterValidator(gvk.String(), DurationValidator)
+}
+
+func EgressValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	egress := &pv1alpha1.Egress{}
+	if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(egress); err != nil {
+		return nil, err
+	}
+
+	for _, m := range egress.Spec.Matches {
+		if m.Kind != "HTTPRouteGroup" {
+			return nil, fmt.Errorf("Egress spec matches kind is %s when it should be HTTPRouteGroup", m.Kind)
```

Comment:
nit: a little less verbose, so maybe rename error message to:

"Expected Matches.Kind to be 'HTTPRouteGroup', got: %s"

and same phrasing for below

Reply:
Updated the error message.